eventManager.js

```diff
@@ -0,0 +1,39 @@
+// Decides what event should be emitted
+// old_status: the statusType of the asset already in the database, is null if the asset has just been added to the database
+// new_asset_status: contains the asset status we just added to the database
+// eventNames: enum containing all the possible events
+// statusTypes: enum containing all the possible types of asset["statusTypes"]
+// returns: event we want to raise
+module.exports = function eventManager(old_status, new_asset_status, eventNames, statusTypes){
+    
+    // Should add error handling if any input parameters are strange
+
+    const new_status = new_asset_status["statusType"]; // for readability
+    let event = null; // want to return null event if none of the events are triggered
+
+    // If new status is normal and the old status is null or not normal    
+    if ((old_status !== statusTypes.NORMAL && new_status === statusTypes.NORMAL) && (old_status !== null && new_status === statusTypes.NORMAL)){
+                
+        event = eventNames.AssetRecovered;             
+    }                    
+
+    // If the status has changed to warning
+    else if (old_status !== statusTypes.WARNING && new_status == statusTypes.WARNING) {

```

Comment:
If there is a need to comment any code it's probably an indicator that the code is not intuitive and can be improved for readability. For example here we could extract the conditions to local functions:
```suggestion
    else if (statusChangedToWarning(new_status, old_status)) {
```
and somewhere above in this module you would declare a local function:
```javascript
const statusChangedToWarning = (new_status, old_status) =>
  old_status !== statusTypes.WARNING && new_status == statusTypes.WARNING
```
In general we are discouraging comments in our code or keep it to minimum for the reasons I explained above.